pkg/controller/integration/integration_controller.go

```diff
@@ -75,6 +77,109 @@ func newReconciler(mgr manager.Manager, c client.Client) reconcile.Reconciler {
 	)
 }
 
+func integrationUpdateFunc(old *v1.Integration, it *v1.Integration) bool {
```

Comment:
+1 for encapsulating these functions. However, I could not understand if there is any logic change in them or it was just the encapsulation which changed. Can you clarify?

Reply:
the linter was complaining about complexity of the method so I created dedicated functions to make the method more readable.